value/src/main/java/com/google/auto/value/processor/autoannotation.vm

```diff
@@ -70,15 +70,11 @@ final class $className implements $annotationName, `java.io.Serializable` {
       $params[$p].type $members[$p] #if ($foreach.hasNext) , #end
 #end ) {
 #foreach ($p in $params.keySet())
-  #if (!$members[$p].kind.primitive)
```

Comment:
I'm not sure I understand why the changes in this file need to be anything more than changing `if (x == null)`+`throw new NPE(msg)` into `Objects.requireNonNull(x, msg)`. And, in fact the opportunity to do so on line 76 seems to have been missed.

Reply:
Refactored.